File: Objetivo 3/HerancaSimplesExemploJava/src/model/Conta.java
package model;

/**
 *
 * @author vagner
 */
public abstract class Conta { //a cláusula abstract impede a criação de instâncias desse classe
    protected double saldo;

    public Conta(){}
    
    public Conta(double valor) {
        this.saldo = valor;
    }
     
    public double getSaldo(){
        return this.saldo;
    }
    
    public void deposita(double valor){
        this.saldo += valor;
     }
    
    public void saca(double valor){ //esse comportamento foi sobrescrito (redefinido) na classe ContaCorrente.
        this.saldo -= valor;
    }
 
    
    public void atualiza(double taxa){
        this.saldo *= (taxa/100);
    }
        
}

File: Objetivo 3/HerancaSimplesExemploPHP/model/Conta.php
<?php

/**
 * Description of Conta
 *
 * @author vagner
 */
abstract class Conta {
    protected $saldo = 0.0;

    public function __construct(){}
     
    public function getSaldo(){
        return $this->saldo;
    }
    
    public function deposita($valor){
        $this->saldo += $valor;
    }
    
    public function saca($valor){ //esse comportamento foi sobrescrito (redefinido) na classe ContaCorrente.
        $this->saldo -= $valor;
    }
 
    
    public function atualiza($taxa){
        $this->saldo *= ($taxa/100);
    }

    public function __toString() {
        return "Conta[saldo=$this->saldo]";
    }
}
